core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -2360,6 +2361,27 @@ private void checkNodeTypeCount(String sql, Map<Class<? extends RelNode>, Intege
     checkNodeTypeCount(sql, expected);
   }
 
+  @Test public void testNodeTypeExchange() {
+
+    final RelNode rel = convertSql("select * from emp");
+    final RelDistribution dist = RelDistributions.hash(ImmutableList.<Integer>of());
+    final LogicalExchange exchange = LogicalExchange.create(rel, dist);
+
+    final Map<Class<? extends RelNode>, Integer> expected = new HashMap<>();
+    expected.put(TableScan.class, 1);
+    expected.put(Exchange.class, 1);
+    expected.put(Project.class, 1);
+
+    final RelMetadataQuery mq = rel.getCluster().getMetadataQuery();
+    final Multimap<Class<? extends RelNode>, RelNode> result = mq.getNodeTypes(exchange);
+    assertThat(result, notNullValue());
+    final Map<Class<? extends RelNode>, Integer> resultCount = new HashMap<>();
+    for (Entry<Class<? extends RelNode>, Collection<RelNode>> e : result.asMap().entrySet()) {
+      resultCount.put(e.getKey(), e.getValue().size());
+    }
+    assertEquals(expected, resultCount);
+  }
+
```

Comment:
Shall we use `checkNodeTypeCount` ?

Reply:
no, exchange cannot be expressed by SQL string.